drivers/gpio/gpio_mcp23xxx.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2021 metraTec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file Header file for the MCP23Xxx driver.
+ */
+
+#ifndef ZEPHYR_DRIVERS_GPIO_GPIO_MCP23XXX_H_
+#define ZEPHYR_DRIVERS_GPIO_GPIO_MCP23XXX_H_
+
+#include <kernel.h>
+
+#include <drivers/gpio.h>
+#ifdef CONFIG_GPIO_MCP230XX
+#include <drivers/i2c.h>
+#endif /* CONFIG_GPIO_MCP230XX */
+#ifdef CONFIG_GPIO_MCP23SXX
+#include <drivers/spi.h>
+#endif /* CONFIG_GPIO_MCP23SXX */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
nit: if only used by the driver (C) this should not be required

Reply:
Yeah, definitely only used by the two drivers, so removed.